### ISSUE TITLE
HasDefaultValue extension method proposes an Identity Column if property is int

### ISSUE THREAD

**Issue by Ponant** [dotnet/efcore#5782]
### Steps to reproduce

Console App (.Net Core)

project.json

```
{
  "version": "1.0.0-*",
  "buildOptions": {
    "emitEntryPoint": true
  },

  "dependencies": {
    "Microsoft.EntityFrameworkCore": "1.0.0-rc2-final",
    "Microsoft.EntityFrameworkCore.SqlServer": "1.0.0-rc2-final",
    "Microsoft.EntityFrameworkCore.Tools": "1.0.0-preview1-final",
    "Microsoft.NETCore.App": {
      "type": "platform",
      "version": "1.0.0-rc2-3002702"
    }
  },
  "tools": {
    "Microsoft.EntityFrameworkCore.Tools": {
      "version": "1.0.0-preview1-final",
      "imports": [
        "portable-net45+win8+dnxcore50"
      ]
    }
  },
  "frameworks": {
    "netcoreapp1.0": {
      "imports": [
        "dnxcore50",
        "portable-net451+win8"
      ]
    }
  }
}

```

using Microsoft.EntityFrameworkCore;

```
    public class Program
    {
        public static void Main(string[] args)
        {

        }
    }
```

```
public class Blog
{
    public int BlogId { get; set; }
    public string Url { get; set; }
    public int Bugging { get; set; }
}
public class AppDbContext : DbContext
{
    public DbSet<Blog> Blogs { get; set; }

    protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
    {
        optionsBuilder.UseSqlServer(@"Server=(localdb)\mssqllocaldb;Database=EFCoreDb;Trusted_Connection=True;");
        //base.OnConfiguring(optionsBuilder);
    }
    protected override void OnModelCreating(ModelBuilder modelBuilder)
    {
        modelBuilder.Entity<Blog>()
           .Property(b => b.Bugging)
           .HasDefaultValue(123);
    }
}
```
### The issue

Following http://ef.readthedocs.io/en/latest/modeling/relational/default-values.html#default-values
I expected the property Bugging to have a default value of 123. However:

add-migration init

in Package Manager Console proposes an  SqlServerValueGenerationStrategy.IdentityColumn to the property Bugging, as seen here

```
public partial class init : Migration
    {
        protected override void Up(MigrationBuilder migrationBuilder)
        {
            migrationBuilder.CreateTable(
                name: "Blogs",
                columns: table => new
                {
                    BlogId = table.Column<int>(nullable: false)
                        .Annotation("SqlServer:ValueGenerationStrategy", SqlServerValueGenerationStrategy.IdentityColumn),
                    Bugging = table.Column<int>(nullable: false, defaultValue: 123)
                        .Annotation("SqlServer:ValueGenerationStrategy", SqlServerValueGenerationStrategy.IdentityColumn),
                    Url = table.Column<string>(nullable: true)
                },
                constraints: table =>
                {
                    table.PrimaryKey("PK_Blogs", x => x.BlogId);
                });
        }
        protected override void Down(MigrationBuilder migrationBuilder)
        {
            migrationBuilder.DropTable(
                name: "Blogs");
        }
    }
```

This will lead to a conflict if we apply update-database "Only one identity column per table is allowed":

Applying migration '20160615215635_init'. System.Data.SqlClient.SqlException: Multiple identity columns specified for table 'Blogs'. Only one identity column per table is allowed. 
   at System.Data.SqlClient.SqlConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at System.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, Boolean callerHasConnectionLock, Boolean asyncClose)
    at System.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj, Boolean& dataReady) 
    at System.Data.SqlClient.SqlCommand.RunExecuteNonQueryTds(String methodName, Boolean async, Int32 timeout, Boolean asyncWrite) 
    at System.Data.SqlClient.SqlCommand.InternalExecuteNonQuery(TaskCompletionSource`1 completion, String methodName, Boolean sendToPipe, Int32 timeout, Boolean asyncWrite) 
   at System.Data.SqlClient.SqlCommand.ExecuteNonQuery()
    at Microsoft.EntityFrameworkCore.Storage.Internal.RelationalCommand.Execute(IRelationalConnection connection, String executeMethod, IReadOnlyDictionary`2 parameterValues, Boolean openConnection, Boolean closeConnection) 
    at Microsoft.EntityFrameworkCore.Storage.Internal.RelationalCommand.ExecuteNonQuery(IRelationalConnection connection, IReadOnlyDictionary`2 parameterValues, Boolean manageConnection) 
    at Microsoft.EntityFrameworkCore.Storage.Internal.RelationalCommandExtensions.ExecuteNonQuery(IEnumerable`1 commands, IRelationalConnection connection) 
    at Microsoft.EntityFrameworkCore.Migrations.Internal.Migrator.Execute(IEnumerable`1 relationalCommands) 
    at Microsoft.EntityFrameworkCore.Migrations.Internal.Migrator.Migrate(String targetMigration) 
    at Microsoft.EntityFrameworkCore.Design.MigrationsOperations.UpdateDatabase(String targetMigration, String contextType) 
    at Microsoft.EntityFrameworkCore.Tools.Cli.DatabaseUpdateCommand.<>c__DisplayClass0_0.<Configure>b__0() 
    at Microsoft.Extensions.CommandLineUtils.CommandLineApplication.Execute(String[] args)
    at Microsoft.EntityFrameworkCore.Tools.Cli.Program.Main(String[] args) 
ClientConnectionId:e82bfd39-4006-47c0-aa74-d76c3e967669
 Error Number:2744,State:2,Class:16
 Multiple identity columns specified for table 'Blogs'. Only one identity column per table is allowed. 
### Notice

This issue is absent if the property Bugging is a string, so it seems EF Core sees 2 ints and proposes them as Id columns.
I had a similar issue without migrations, using only db.DataBase.EnsureCreated() in a using statement in Main (not shown)
Any advice is welcomed.
### Further technical details

Operating system: W10 pro 64 bit
Visual Studio version: VS 2015 Update 3 RC
### Update

This will not work neither
`.ForSqlServerHasDefaultValue(567);
`
Adding a third int property leads to 3 identity columns when using HasDefaultValue.

This works
`.HasDefaultValueSql("123");`

Another way to make it work is to set a default value in the Blog ctor, or inline as such

`public int Rating { get; set; } = 333;`

and delete the fluent api code. Actually, this makes me wonder what would be the advantage off using HasDefaultValue ?

**Comment by rowanmiller** [comment]
We have fixed some issues around this functionality. What version of EF Core are you using?

**Comment by Ponant** [comment]
"Microsoft.EntityFrameworkCore": "1.0.0-rc2-final"
Is that what you want to know?

**Comment by ajcvickers** [comment]
Verified fixed by #5384
